Sources/Parser.swift

```diff
@@ -40,16 +75,37 @@ public enum Result<SuccessType> {
     }
 }
 
+/**
+ Instances of this class parse XML documents into a tree of DOM objects.
+ */
 public class Parser {
     private let parser: XMLParser
 
     private let log = Log(level: .all)
 
+    /**
+     Used to represent a parser error.
+     */
     public enum Error: Swift.Error {
-        case invalidXML
+        /**
+         Indicates that the parser terminated abnormally resulting in the 
+         associated error value
+         */
+        case parserError(Swift.Error)
+
+        /**
+         Indicates that the parser terminated abnormally but did not report an 
+         error.
+         */
         case unspecifiedError
     }
 
+    /**
+     Initializes a parser with the XML content referenced by the given URL.
+     
+     - parameter url: A `URL` object specifying a URL. The URL must be fully qualified and refer to a scheme that is supported by the `URL` type.
+     - returns: An initialized `Parser` object or `nil` if an error occurs.
+     */
```

Comment:
Add a parameter to all initializers for processing namespaces, default to false.